robottelo/rhsso_utils.py

```diff
@@ -0,0 +1,121 @@
+"""Utility module to handle the rhsso-satellite configure UI/CLI/API testing"""
+import json
+import random
+
+from robottelo import ssh
+from robottelo.cli.base import CLIReturnCodeError
+from robottelo.config import settings
+from robottelo.constants import KEY_CLOAK_CLI
+from robottelo.constants import RHSSO_NEW_USER
+from robottelo.constants import RHSSO_RESET_PASSWORD
+from robottelo.datafactory import gen_string
+from robottelo.datafactory import valid_emails_list
+
+
+def run_command(cmd, hostname=settings.server.hostname, timeout=None):
+    """helper function for ssh command and avoiding the return code check in called function"""
+    if timeout is not None:
+        result = ssh.command(cmd=cmd, hostname=hostname, timeout=timeout)
+    else:
+        result = ssh.command(cmd=cmd, hostname=hostname)
+    if result.return_code != 0:
+        raise CLIReturnCodeError(
+            result.return_code, result.stderr, 'Failed to run the command : {}'.format(cmd),
+        )
+    else:
+        return result.stdout
+
+
+def get_rhsso_client_id():
+    """Getter method for fetching the client id and can be used other functions"""
+    client_name = "{}{}".format(settings.server.hostname, '-foreman-openidc')
+    run_command(
+        cmd="{0} config credentials "
+        "--server {1}/auth "
+        "--realm {2} "
+        "--user {3} "
+        "--password {4}".format(
+            KEY_CLOAK_CLI,
+            settings.rhsso.host_url.replace("https://", "http://"),
+            settings.rhsso.realm,
+            settings.rhsso.rhsso_user,
+            settings.rhsso.password,
+        ),
+        hostname=settings.rhsso.host_name,
+    )
+
+    result = run_command(
+        cmd="{} get clients --fields id,clientId".format(KEY_CLOAK_CLI),
+        hostname=settings.rhsso.host_name,
```

Comment:
Repetative use of hostname settings. store in var and reuse. same applies for other settings if any.

Reply:
Can you please give me an example robottelo code. How this will be helpful?

Reply:
Okay I have done this at module level

Reply:
Cool. In feature PRs you can try using pytest fixture capabilities, though it's not going to speed up the process.

Like:
```
@pytest.fixture(scope='module')
def rhsso_settings():
   return {'satellite': settings.server.hostname, 'rhsso_host' : settings.rhsso.host_name ...... }
```

Call this function in test functions and u shuld be able to use settings as rhsso_settings['satellite'] under test.

Not blocking your PR for these improvements.